### PR TITLE
Add `.nullable` property

### PR DESCRIPTION
**Pull request by jhoermann**
## Why
Props may also be `null` and not `undefined` depending on the model. Using the `.optional` property would then lead to a type error: `Type 'T | null' is not assignable to type 'T | undefined'.`.

## How
- [x] Added prop type function `nullable`
- [x] Added `nullable` to prop options generator
- [x] Added tests
- [x] Changed docs

### CODE REVIEW

**Comment by jhoermann** [comment]
I think I'm missing some changes in the readme file - will add them.

File: src/prop-types/function.ts [hunk 1]
```diff
@@ -23,6 +24,13 @@ export const functionProp = <T extends Function>(validator?: Validator): Functio
     default: undefined,
     validator: vuePropValidator(validator),
   },
+  nullable: {
+    type: Function,
+    required: false,
+    // eslint-disable-next-line unicorn/no-null
```
Comment (FloEdelmann) [inline src/prop-types/function.ts:30]: I think we can disable the rule globally

Comment (jhoermann) [reply src/prop-types/function.ts]: Done now in https://github.com/FloEdelmann/vue-ts-types/pull/89/commits/433f10686e1d9fd72508e4049d6cce21f7ad0308

**Comment by jhoermann** [comment]
> I think I'm missing some changes in the readme file - will add them.

Updated it now.

File: src/prop-types/function.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { vuePropValidator } from '../validators';
 
 interface FunctionPropOptionsGenerator<T> {
   optional: DefaultPropOptions<T | undefined> & { default?: () => T };
+  nullable: DefaultPropOptions<T | null> & { default?: (() => T) | null };
```
[the remark this answers is not in the document]
Comment (jhoermann) [reply src/prop-types/function.ts:7]: Tests run through now.

**Review by FloEdelmann — APPROVED** [review]
Thanks for updating the tests and documentation!

This looks really good :)

**Pull request merged**